TF_main_loop/main.py

```diff
@@ -306,9 +306,12 @@ def __run(build_model):
                     0,
                     which_set=s,
                     stateful_validation=cfg.stateful_validation,
-                    save_samples=True,
-                    save_heatmap=True,
-                    save_raw_predictions=False)
+                    save_samples=cfg.save_samples,
```

Comment:
`cfg` is a global variable, you don't need to pass these arguments as a parameter to the function.
Since you are there, please remove `stateful_validation` as well! :D